### DISCUSSION TITLE
Redis는 정말 빠를까?

### DISCUSSION THREAD

**Discussion by kyungjunleeme** [kyungjunleeme/kafka_study#52] [Q&A]
https://siyul-park.github.io/database/is-redis-really-fast/

![image](https://github.com/user-attachments/assets/28aeb2d2-b72d-41da-ad82-cd700c26df26)

**Comment by hellices** [comment]
redis 한국 지사가 생겨서 linkedin에 많은 자료를 올리고 있습니다.
아래는 redis 엔지니어가 작성한 내용입니다.

Which latency numbers you should know?

Please note those are not accurate numbers. They are based on some online benchmarks (Jeff Dean’s latency numbers + some other sources).

🔹L1 and L2 caches: 1 ns, 10 ns
E.g.: They are usually built onto the microprocessor chip. Unless you work with hardware directly, you probably don’t need to worry about them.

🔹RAM access: 100 ns
E.g.: It takes around 100 ns to read data from memory. Redis is an in-memory data store, so it takes about 100 ns to read data from Redis.

🔹Send 1K bytes over 1 Gbps network: 10 us
E.g.: It takes around 10 us to send 1KB of data from Memcached through the network.

🔹Read from SSD: 100 us
E.g.: RocksDB is a disk-based K/V store, so the read latency is around 100 us on SSD.

🔹Database insert operation: 1 ms.
E.g.: Postgresql commit might take 1ms. The database needs to store the data, create the index, and flush logs. All these actions take time.

🔹Send packet CA->Netherlands->CA: 100 ms
E.g.: If we have a long-distance Zoom call, the latency might be around 100 ms.

🔹Retry/refresh internal: 1-10s
E.g: In a monitoring system, the refresh interval is usually set to 5~10 seconds (default value on Grafana).

Notes
-----
1 ns = 10^-9 seconds
1 us = 10^-6 seconds = 1,000 ns
1 ms = 10^-3 seconds = 1,000 us = 1,000,000 ns

--
Subscribe to our weekly newsletter to get a Free System Design PDF (158 pages): https://bit.ly/3KCnWXq

![image](https://github.com/user-attachments/assets/ead5a17d-2aa9-4f21-ba96-0b5a0935f2af)

**Comment by kyungjunleeme** [comment]
@hellices 한국지사도 생겼군요 ㅎㅎ몰랐습니다.

https://www.linkedin.com/in/bonghwan-kim-9a30701/

알아보니 이 분이 한국지사에서 근무하시는 분이 더라구요.

https://university.redis.io/  # 공식 교육 사이트도 있네요.

https://www.facebook.com/groups/rediskorea/
여기도 좋은 글이 많네요.

https://docs.google.com/presentation/d/1z3mV0mwTL4tqa492f5e_QotIZA7ouWL9-aAffRWq0u8/edit?fbclid=IwZXh0bgNhZW0CMTEAAR2SdnWjoR_K5FHXICNbmPhXq8lDBsIZhNl_p9uoxe1SIIxqpeoPFYMQP7o_aem_mquWRsS5PckEPlZ_BsC4Ag#slide=id.p

https://docs.google.com/presentation/d/193Bra3z7kjhx1YcrIzo9oxcuJ5lZlqv_0WWEiO2TAuM/edit?fbclid=IwZXh0bgNhZW0CMTAAAR2-QR2JCbFgrMpnN6N4Vnj9vWeTkEKwNhAuqJWy1C1G2Xx93bPFq3r1F2s_aem_UHOzAXQc_5mh11c73cilWw#slide=id.g2cbeaff51fd_0_98

**Comment by kyungjunleeme** [comment]
![image](https://github.com/user-attachments/assets/3f38d315-817d-49fc-bfa5-13b17cb0ebc6)

The diagram below shows 3 main reasons: 
 
1️⃣ Redis is RAM-based. Access to RAM is at least 1000 times faster than access to random disks. Redis can be used as a cache to improve application responsiveness and reduce database load when used as a cache. 
 
2️⃣ Redis implements IO multiplexing and single-threaded execution. 
 
3️⃣ A number of efficient lower-level data structures are leveraged by Redis. By keeping these data structures in memory, serialization and deserialization costs are reduced. 
 
SortedSet, for example, makes the implementation of leaderboards so simple and efficient. On the other hand, a bitmap can be used to aggregate month-over-month login statuses. 
 
Over to you: What do you use Redis for in your projects? 

—
Subscribe to our weekly newsletter to get a Free System Design PDF (158 pages): https://bit.ly/bbg-social

[해시태그#systemdesign](https://www.linkedin.com/feed/hashtag/?keywords=systemdesign&highlightedUpdateUrns=urn%3Ali%3Aactivity%3A7286639745001009152) [해시태그#coding](https://www.linkedin.com/feed/hashtag/?keywords=coding&highlightedUpdateUrns=urn%3Ali%3Aactivity%3A7286639745001009152) [해시태그#interviewtips](https://www.linkedin.com/feed/hashtag/?keywords=interviewtips&highlightedUpdateUrns=urn%3Ali%3Aactivity%3A7286639745001009152)
